lib/iris/tests/unit/fileformats/netcdf/test_Saver__ugrid.py

```diff
@@ -0,0 +1,1171 @@
+            if result:
+                # val='*'' for a simple existence check
+                result = (val == "*") or attrs[key] == val
+            if not result:
+                break
+        return result
+
+    varsdict = {
+        name: attrs
+        for name, attrs in varsdict.items()
+        if check_attrs_match(attrs)
+    }
+    return varsdict
+
+
+def vars_w_dims(varsdict, dim_names):
+    """Subset a vars dict, returning those which map all the specified dims."""
+    varsdict = {
+        name: propsdict
+        for name, propsdict in varsdict.items()
+        if all(dim in propsdict["_DIMS"] for dim in dim_names)
+    }
+    return varsdict
+
+
+def vars_meshnames(vars):
+    """Return the names of all the mesh variables (found by cf_role)."""
+    return list(vars_w_props(vars, cf_role="mesh_topology").keys())
+
+
+def vars_meshdim(vars, location, mesh_name=None):
+    """
+    Extract a dim-name for a given element location.
+
+    Args:
+        * vars (varsdict):
+            file varsdict, as returned from 'snapshot_dataset'.
+        * location (string):
+            a mesh location : 'node' / 'edge' / 'face'
+        * mesh_name (string or None):
+            If given, identifies the mesh var.
+            Otherwise, find a unique mesh var (i.e. there must be exactly 1).
+
+    Returns:
+        dim_name (string)
+            The dim-name of the mesh dim for the given location.
+
+    TODO: relies on the element having coordinates, which in future will not
+        always be the case. This can be fixed
+
+    """
+    if mesh_name is None:
+        # Find "the" meshvar -- assuming there is just one.
+        (mesh_name,) = vars_meshnames(vars)
+    mesh_props = vars[mesh_name]
+    loc_coords = mesh_props[f"{location}_coordinates"].split(" ")
+    (single_location_dim,) = vars[loc_coords[0]]["_DIMS"]
+    return single_location_dim
+
+
+class TestSaveUgrid__cube(tests.IrisTest):
+    """Test for saving cubes which have meshes."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.temp_dir = Path(tempfile.mkdtemp())
+
+    @classmethod
+    def tearDownClass(cls):
+        shutil.rmtree(cls.temp_dir)
+
+    def check_save_cubes(self, data):
+        """
+        Write cubes to a new file in the common temporary directory.
+
+        Use a name unique to this testcase, to avoid any clashes.
+
+        """
+        # use 'result_path' to name the file after the test function
+        tempfile_path = self.result_path(ext=".nc")
+        # Create a file of that name, but discard the result path and put it
+        # in the common temporary directory.
+        tempfile_path = self.temp_dir / Path(tempfile_path).name
+
+        # Save data to the file.
+        save(data, tempfile_path)
+
+        return tempfile_path
+
+    def test_basic_mesh(self):
+        # Save a small mesh example and check aspects of the resulting file.
+        data = make_cube()  # A simple face-mapped data example.
```

Comment:
`data` -> `cube` ? (and below, if you agree)

Reply:
:+1: